### PR TITLE
#15: tools for work with git 

### PR DESCRIPTION
**Pull request by katridi**
Here is my first try implementing git workflow.
Maybe it's worth to implement new class since with all func repo is required
Any suggestions, critics, nit picking are higly accepted

### CODE REVIEW

**Comment by apatrushev** [comment]
I suggest to see on provided API in first place but not on implementation details so we can easily replace implementation later.

File: work_with_git.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import git
+from git import Repo
+from git import Actor
+
+
+author = Actor("An author", "katridi@yandex.ru")
+committer = Actor("A committer", "katridi@yandex.ru")
+
+def clone_repo(url, path):
```
Comment (apatrushev) [inline work_with_git.py]: Looks like we are using repo as first argument in all methods besides this, so we can create our own proxy class and convert clone to factory `classmethod`. Pseudo-code below:
```python
class Repo:
    @classmethod
    def clone(cls, path):
        .... returning instance of cls
```

Comment (katridi) [reply work_with_git.py]: Thanks, I'll consider it.

Comment (katridi) [reply work_with_git.py]: For now I'll wait for others, but your suggestion seems reasonable enough!

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+GitPython
```
Comment (mjecov) [inline requirements.txt]: Please, fix current library version in requirements and add blank new line in the end

**Review by mjecov — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (katridi) [reply requirements.txt]: Yes, thanks

File: work_with_git.py (hunk 1 above)
Comment (katridi) [reply work_with_git.py]: Could you explain to me and possibly to others why @classmethod is more preferable than this snipet:
https://gist.github.com/jekoff/3ce51119a76b01345d97c72da99c1c52
Here one initializes self.repo and then do whatever he need
Thanks in advance!

Comment (apatrushev) [reply work_with_git.py]: It is better to keep `__init__` method as simple as possible. So any actions beyond the object construction itself should be separated. It is my personal opinion for sure. Just compare usage snippets:
```python
repo = Repository('git@github.com:sphericalpm/lektorium.git', '/tmp/lektorium')
```
```python
repo = Repository.clone('git@github.com:sphericalpm/lektorium.git', '/tmp/lektorium')
```
In second example you clearly show the action to be taken. Explicit is better than implicit!

Comment (katridi) [reply work_with_git.py]: Fair enough, thanks!
I'll work on it

**Comment by AndreiDrang** [comment]
Waiting doc-strings and comments in methods and class description.
+ Type hints for variables and methods.

**Pull request merged**